staff_features/resources/step_definitions/resource_tree_navigate.rb

```diff
@@ -0,0 +1,141 @@
+  select 'Class', from: 'resource_level_'
+  element = find('#resource_lang_materials__0__language_and_script__language_')
+  element.send_keys('AU')
+  element.send_keys(:tab)
+
+  select 'Single', from: 'resource_dates__0__date_type_'
+  fill_in 'resource_dates__0__begin_', with: '2024'
+
+  fill_in 'resource_extents__0__number_', with: '10'
+  select 'Cassettes', from: 'resource_extents__0__extent_type_'
+
+  element = find('#resource_finding_aid_language_')
+  element.send_keys('ENG')
+  element.send_keys(:tab)
+
+  element = find('#resource_finding_aid_script_')
+  element.send_keys('Latin')
+  element.send_keys(:tab)
+
+  find('button', text: 'Save Resource', match: :first).click
+
+  wait_for_ajax
+  expect(page).to have_text "Resource Resource #{@uuid} created"
+
+  url_parts = current_url.split('/')
+  url_parts.pop
+  @resource_id = url_parts.pop
+
+  click_on 'Add Child'
+  wait_for_ajax
+  fill_in 'Title', with: "Archival Object 1 #{@uuid}"
+  select 'Class', from: 'Level of Description'
+  click_on 'Save'
+  expect(page).to have_text "Archival Object Archival Object 1 #{@uuid} on Resource Resource #{@uuid} created"
+
+  click_on 'Add Child'
+  wait_for_ajax
+  fill_in 'Title', with: "Archival Object 2 #{@uuid}"
+  select 'Class', from: 'Level of Description'
+  click_on 'Save'
+  expect(page).to have_text "Archival Object Archival Object 2 #{@uuid} created as child of Archival Object 1 #{@uuid} on Resource Resource #{@uuid}"
+  wait_for_ajax
+end
+
+Given 'the Resource is opened in edit mode' do
+  visit "#{STAFF_URL}/resources/#{@resource_id}/edit"
+
+  wait_for_ajax
+end
+
+When 'the user opens the Resource in edit mode' do
+  visit "#{STAFF_URL}/resources/#{@resource_id}/edit"
+
+  wait_for_ajax
+end
+
+Then 'the Resource is displayed as the top level of the navigation tree' do
+  rows = all('#tree-container .table .table-row', visible: :all)
+
+  expect(rows.length).to eq 4
+  expect(rows[0].text).to include "Resource #{@uuid}"
+end
+
+Then 'the tree includes the two Archival Objects' do
+  rows = all('#tree-container .table .table-row')
+
+  expect(rows.length).to eq 2
+  expect(rows[2].text).to include "Archival Object 1 #{@uuid}"
+end
+
+Then 'the Resource is highlighted in the tree' do
+  rows = all('#tree-container .table .table-row')
+
+  expect(rows.length).to eq 2
+  expect(rows[0].text).to include "Resource #{@uuid}"
+  expect(rows[0][:class]).to include 'current'
+end
+
+Given 'only the first-level Archival Objects are displayed' do
+  rows = all('#tree-container .table .table-row')
+
+  expect(rows.length).to eq 2
+  expect(rows[1].text).to include "Archival Object 1 #{@uuid}"
+end
+
+Then 'the button has text {string}' do |button_text|
+  expect(page).to have_text button_text
+end
+
+Then 'the expand arrows are disabled' do
+  arrows = all('.expandme')
+
+  arrows.each do |arrow|
+    expect(arrow[:class]).to include 'disabled'
+  end
+end
+
+Then 'all Archival Objects are displayed' do
+  wait_for_ajax
```

Comment:
I feel like there needs to be more consistency generally as to when the wait_for_ajax is called. Sometimes it is right after navigating, and then sometimes its before a test, like here. It should probably be closest to whatever action kicks off the ajax calls.

Reply:
Generally, I agree, however, this is not always possible. The end goal is to have as many reusable steps as possible. It has already occurred, that a new feature didn't require any step definitions because it was already covered by previous ones. The previous step, in this case, is `When 'the user clicks on {string}' do |string|`, and since `all Archival Objects are displayed` is always through an AJAX call, I think it is ok to have it here.

Maybe we can move the `wait_for_ajax` inside `When 'the user clicks on {string}' do |string|` in the `shared/step_definitions.rb`. I think I have already done this on another PR. Note that `wait_for_ajax` is actually pretty fast, if there are no requests running, which is usually the case, so I believe it is totally harmless to use it anywhere applicable, to avoid flakiness.

@balmas  @thimios @brianzelip 
What are your thoughts on this? Do you have in mind any implications or downsides by making extensive or consecutive use of `wait_for_ajax`, to avoid frequently failing tests?

Reply:
Other than possibly adding more time to how long the tests take, I don't see any problems with it. It is common to use such a pattern to work against the flakiness that we so often experience with testing ArchivesSpace.